### PR TITLE
Configurable detection levels in ISO14443A and Legic modes

### PR DESCRIPTION
**Pull request by czietz**
As recently discussed on Discord, the received signal on my PM3 Easy is [too noisy](https://cdn.discordapp.com/attachments/738516015297593445/1202350784696828024/image.png) for the hard-coded detection/demodulation thresholds. I assume this affects all devices where [the PCB looks like this.](https://cdn.discordapp.com/attachments/738516015297593445/1202315885600329738/IMG_5992.JPG)

With the official firmware ISO14443A and Legic tags are thus not detected. This PR adds a commands `hw sethfthresh` to configure the thresholds at runtime. The default thresholds remain unchanged.

### CODE REVIEW

File: client/src/cmdhw.c [hunk 1]
```diff
@@ -743,6 +743,39 @@ static int CmdSetDivisor(const char *Cmd) {
...
+    params[2] = arg_get_int_def(ctx, 3, 8);
+    CLIParserFree(ctx);
+
+    if ((params[0] < 1) || (params[0] > 63) || (params[1] < 1) || (params[1] > 63)) {
+        PrintAndLogEx(ERR, "Thresholds must be between " _YELLOW_("1") " and " _YELLOW_("63"));
+        return PM3_EINVARG;
+    }
+
+    clearCommandBuffer();
+    SendCommandNG(CMD_HF_ISO14443A_SET_THRESHOLDS, (uint8_t *)&params, sizeof(params));
```
Comment (iceman1001) [inline client/src/cmdhw.c:774]: The preferred way is to use a struct.   
Makes things more easier to understand and follow.

Comment (czietz) [reply client/src/cmdhw.c:774]: I'll change that.

File: fpga/fpga_pm3_top.v [hunk 2]
```diff
@@ -111,6 +111,8 @@ always @(posedge spck) if (~ncs) shift_reg <= {shift_reg[14:0], mosi};
 reg trace_enable;
 
 reg [7:0] lf_ed_threshold;
+reg [10:0] hf_edge_detect_threshold;
+reg [10:0] hf_edge_detect_threshold_high;
```
[the remark this answers is not in the document]
Comment (czietz) [reply fpga/fpga_pm3_top.v:115]: It's a design philosophy thing. In my HDL designs, I try to be consistent with bit lengths all the way through. Down in one compares the thresholds against `rx_mod_falling_edge_max` and `rx_mod_rising_edge_max` which are `[10:0]`. Note that the compiler will not synthesize unused bits into flip-flops/latches anyway.

But you are right, it is not necessary. I'll change it.

[the remark this answers is not in the document]
Comment (iceman1001) [reply fpga/fpga_pm3_top.v]: I can't say anything about what the compiler will do or not,   but the fpga is old and scarce with resources.

**Pull request merged**